packages/core/src/sync-realtime/service.ts

```diff
@@ -80,6 +80,14 @@ export class RealtimeSyncService extends Emittery<RealtimeSyncEvents> {
     const latestBlock = await this.getLatestBlock();
     const latestBlockNumber = hexToNumber(latestBlock.number);
 
+    // Fetch the chainId from the rpc, warn if not matching
+    const rpcChainId = await this.network.client.getChainId();
```

Comment:
Would be best to do these requests concurrently, because this is in the critical path during setup.